packages/flutter_tools/lib/src/commands/build_ios.dart

```diff
@@ -463,13 +463,15 @@ class BuildIOSArchiveCommand extends _BuildIOSSubCommand {
     final String relativeOutputPath = app.ipaOutputPath;
     final String absoluteOutputPath = globals.fs.path.absolute(relativeOutputPath);
     final String absoluteArchivePath = globals.fs.path.absolute(app.archiveBundleOutputPath);
-    final String exportMethod = stringArg('export-method')!;
+    String? exportOptions = exportOptionsPlist;
+    final String exportMethod = exportOptions != null
+          ? globals.plistParser.getValueFromFile(exportOptions, 'method') ?? 'unknown'
+          : stringArg('export-method')!;
     final bool isAppStoreUpload = exportMethod  == 'app-store';
```

Comment:
Is `unknown` a reasonable default here? Looking at the output of `xcodebuild -h`, `unknown` is not a valid value.

Should we changing the default at all? Shouldn't we use the existing default of `stringArg('export-method')`?

```suggestion
    final String? exportMethodFromExportOptionsPlist = exportOptions != null
        ? globals.plistParser.getValueFromFile(exportOptions, 'method')
        : null;
    final String exportMethod = exportMethodFromExportOptionsPlist ?? stringArg('export-method')!;
    final bool isAppStoreUpload = exportMethod == 'app-store';
```

Reply:
I get what this is doing, because `exportOptionsPlist` and `export-method` are validated to not both be passed in.  The `unknown` is confusing though.
How about:

```suggestion
    String? exportMethod = exportOptions != null ?
        globals.plistParser.getValueFromFile<String>(exportOptions, 'method') : null;
    exportMethod ??= stringArg('export-method')!;
    final bool isAppStoreUpload = exportMethod == 'app-store';
```
Dart will promote the local to a non-null `String`.